test/secure_store_test.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:authenticator/api/secure_store.dart';
+import 'package:authenticator/model/account.dart';
+
+class FlutterSecureStorageMock extends FlutterSecureStorage {
+  final Map<String, String> storage = {};
+
+  @override
+  Future<String?> read({required String key, IOSOptions? iOptions, AndroidOptions? aOptions, LinuxOptions? lOptions, WebOptions? webOptions, MacOsOptions? mOptions, WindowsOptions? wOptions}) async {
+    String? value = storage[key];
+    return value;
+  }
+
+  @override
+  Future<void> write({required String key, required String? value, IOSOptions? iOptions, AndroidOptions? aOptions, LinuxOptions? lOptions, WebOptions? webOptions, MacOsOptions? mOptions, WindowsOptions? wOptions}) async {
+    if (value != null) {
+      storage[key] = value;
+    }
+  }
+}
+
+void main() {
+  Account accountOne = Account(uuid: "12345");
+  Account accountTwo = Account(uuid: "abcde");
+
+  test('secure store starts empty', () async {
+    SecureStore store = SecureStore(FlutterSecureStorageMock());
+
+    List<Account> accounts = await store.getAccounts();
+
+    expect(accounts.length, 0);
+  });
+
+  test('accounts can be added to secure store', () async {
+    SecureStore store = SecureStore(FlutterSecureStorageMock());
+
+    await store.addAccount(accountOne);
+    await store.addAccount(accountTwo);
+
+    List<Account> accounts = await store.getAccounts();
+
+    expect(accounts.length, 2);
+    expect(accounts[0].uuid, accountOne.uuid);
+    expect(accounts[1].uuid, accountTwo.uuid);
+  });
+
+  test('duplicate accounts should not be added to secure store', () async {
+    SecureStore store = SecureStore(FlutterSecureStorageMock());
+
+    await store.addAccount(accountOne);
+    await store.addAccount(accountOne);
+    await store.addAccount(accountTwo);
+
+    List<Account> accounts = await store.getAccounts();
+
+    expect(accounts.length, 2);
+  });
```

Comment:
Do we want to throw an exception if an account already exists? That way we can more easily catch bugs that do that if they come up.

Reply:
That is a good idea. Silently failing a duplicate insertion could make troubleshooting difficult down the line.

We could also consider overwriting the old entry on a duplicate addition (though I think this functionality would be better to include in a separate function 'updateAccount').

Reply:
I like your second idea, having a separate function whose express purpose is to duplicate sounds like a good plan.